std/algorithm/searching.d

```diff
@@ -4104,7 +4104,24 @@ if (isInputRange!Range && Needles.length > 1 &&
     is(typeof(.startsWith!pred(doesThisStart, withOneOfThese[0])) : bool ) &&
     is(typeof(.startsWith!pred(doesThisStart, withOneOfThese[1 .. $])) : uint))
 {
-    alias haystack = doesThisStart;
+    import std.meta : allSatisfy;
+
+    template checkType(T)
+    {
+        enum checkType = is(Unqual!(ElementEncodingType!Range) == Unqual!T);
+    }
+
+    // auto-decoding special case
+    static if (__traits(isSame, binaryFun!pred, binaryFun!"a == b") &&
```

Comment:
Use `alias pred = "a == b"` to `(a, b) => a == b` and then you can use here `(a, b) => a == b`
No need to stress the compiler more than necessary with mixins.